File: src/main/java/com/protryon/jasm/stage1/constant/CClass.java
package com.protryon.jasm.stage1.constant;

import java.io.DataOutputStream;
import java.io.IOException;

import com.protryon.jasm.stage1.CConstant;

public class CClass extends CConstant<Integer> {
	public CClass(int ref) {
		super(ref, 7);
	}

	@Override
	public void write(DataOutputStream out) throws IOException {
		out.writeShort(this.value);
	}

	public static CClass assertCClass(CConstant CConstant) {
		if (!(CConstant instanceof CClass)) {
			throw new RuntimeException("Expected CClass!");
		}
		return (CClass) CConstant;
	}
	
}

File: src/main/java/com/protryon/jasm/stage1/constant/CClass.kt
package com.protryon.jasm.stage1.constant

import java.io.DataOutputStream
import java.io.IOException

import com.protryon.jasm.stage1.CConstant

class CClass(ref: Int) : CConstant<Int>(ref, 7) {

    @Throws(IOException::class)
    public override fun write(out: DataOutputStream) {
        out.writeShort(this.value)
    }

    companion object {

        fun assertCClass(CConstant: CConstant<*>): CClass {
            if (CConstant !is CClass) {
                throw RuntimeException("Expected CClass!")
            }
            return CConstant
        }
    }

}
